### DISCUSSION TITLE
Does Mastodon support OAuth 2.0 PKCE?

### DISCUSSION THREAD

**Discussion by johnspurlock** [mastodon/mastodon#17207] [Q&A] [answered]
Passing the pkce oauth params `code_challenge` and `code_challenge_method` to [/oauth/authorize](https://docs.joinmastodon.org/methods/apps/oauth/#authorize-a-user) appears to have no effect.

When calling the subsequent [/oauth/token](https://docs.joinmastodon.org/methods/apps/oauth/#obtain-a-token) to obtain a token, the `code_verifier` is apparently not verified.

Server version 3.4.4

**Comment by nightpool** [comment, marked as answer]
No, Doorkeeper does not enable PKCE by default, and Mastodon has not done so.

**Reply by johnspurlock** [reply]
Can you consider making it available?  It's an important security improvement and should not break existing clients, since clients need to opt in via the additional `code_challenge` and `code_challenge_method` params to `/oauth/authorize`

**Reply by ilovecomputers** [reply]
@johnspurlock I created an issue to turn on PKCE https://github.com/mastodon/mastodon/issues/21913

However, for my particular use case of creating a browser-only client, I deemed it unnecessary, but add on the need for your use case.

**Reply by johnspurlock** [reply]
thank you!
